spokestack/mic/pyaudio.py

```diff
@@ -0,0 +1,85 @@
+"""
+This module uses pyaudio to receive input from a
+device microphone
+"""
+import pyaudio  # type: ignore
+
+
+class PyAudioMicrophoneInput:
+    """ This class retrieves audio from an input device
+
+    :param sample_rate: desired sample rate for input
+    :type sample_rate: int
+    :param frame_width: desired frame width for input
+    :type frame_width: int
+    :param exception_on_overflow: produce exception for input overflow
+    :type exception_on_overflow: bool
+    """
+
+    def __init__(
+        self, sample_rate: int, frame_width: int, exception_on_overflow: bool = True
+    ) -> None:
+        self._sample_rate = sample_rate
+        self._frame_size = int(sample_rate / 1000 * frame_width)
+        self._execption_on_overflow = exception_on_overflow
+        self._audio = pyaudio.PyAudio()
+        device = self._audio.get_default_input_device_info()
+        self._stream = self._audio.open(
+            input=True,
+            input_device_index=device["index"],
+            format=pyaudio.paInt16,
+            frames_per_buffer=self._frame_size,
+            channels=1,
+            rate=self._sample_rate,
+            start=False,
+        )
+
+    def __call__(self) -> bytes:
```

Comment:
I think making speech pipeline components callable makes sense, but not for inputs like this
`mic.read()` reads much better than `mic()` and fits most the expectation for the method to call on an input object

Reply:
Yeah I agree. I don't know why that seemed like a good idea yesterday for this class.